command/head.go

```diff
@@ -0,0 +1,313 @@
+
+	2. Check if a remote bucket exists
+		 > s5cmd {{.HelpName}} s3://bucket 
+	
+	3. Print a remote object's metadata with human-readable sizes
+		 > s5cmd {{.HelpName}} --humanize s3://bucket/prefix/object
+	
+	4. Print a remote object's metadata with ETag
+		 > s5cmd {{.HelpName}} --etag s3://bucket/prefix/object
+	
+	5. Print a remote object's fullpath
+		 > s5cmd {{.HelpName}} --show-fullpath s3://bucket/prefix/object
+	
+	6. Print a remote object's metadata with version ID
+		 > s5cmd {{.HelpName}} --version-id VERSION_ID s3://bucket/prefix/object
+	
+	7. Print a remote object's metadata with raw input
+		 > s5cmd {{.HelpName}} --raw s3://bucket/prefix/object
+
+`
+
+func NewHeadCommand() *cli.Command {
+	cmd := &cli.Command{
+		Name:     "head",
+		HelpName: "head",
+		Usage:    "print remote object metadata",
+
+		CustomHelpTemplate: headHelpTemplate,
+
+		Flags: []cli.Flag{
+			&cli.BoolFlag{
+				Name:    "etag",
+				Aliases: []string{"e"},
+				Usage:   "show entity tag (ETag) in the output",
+			},
+			&cli.BoolFlag{
+				Name:    "humanize",
+				Aliases: []string{"H"},
+				Usage:   "human-readable output for object sizes",
+			},
+			&cli.BoolFlag{
+				Name:    "storage-class",
+				Aliases: []string{"s"},
+				Usage:   "display full name of the object class",
+				Value:   true,
+			},
+			&cli.BoolFlag{
+				Name:  "show-fullpath",
+				Usage: "shows only the fullpath names of the object(s)",
+			},
+			&cli.StringFlag{
+				Name:  "version-id",
+				Usage: "use the specified version of an object",
+			},
+			&cli.BoolFlag{
+				Name:  "raw",
+				Usage: "disable the wildcard operations, useful with filenames that contains glob characters",
+			},
+		},
+
+		Before: func(c *cli.Context) error {
+			err := validateHeadCommand(c)
+			if err != nil {
+				printError(commandFromContext(c), c.Command.Name, err)
+			}
+			return err
+		},
+		Action: func(c *cli.Context) (err error) {
+			defer stat.Collect(c.Command.FullName(), &err)()
+
+			op := c.Command.Name
+			fullCommand := commandFromContext(c)
+			src, err := url.New(c.Args().Get(0), url.WithVersion(c.String("version-id")),
+				url.WithRaw(c.Bool("raw")))
+			if err != nil {
+				printError(fullCommand, op, err)
+				return err
+			}
+
+			return Head{
+				src:         src,
+				op:          op,
+				fullCommand: fullCommand,
+				// flags
+				showEtag:         c.Bool("etag"),
+				humanize:         c.Bool("humanize"),
+				showStorageClass: c.Bool("storage-class"),
+				showFullPath:     c.Bool("show-fullpath"),
+
+				storageOpts: NewStorageOpts(c),
+			}.Run(c.Context)
+		},
+	}
+	cmd.BashComplete = getBashCompleteFn(cmd, true, false)
+	return cmd
+}
+
+type Head struct {
+	src         *url.URL
+	op          string
+	fullCommand string
+
+	showEtag         bool
+	humanize         bool
+	showStorageClass bool
+	showFullPath     bool
+
+	storageOpts storage.Options
+}
+
+func (h Head) Run(ctx context.Context) error {
+	h.src.SetRelative(h.src)
```

Comment:
what is the purpose of this call? could you please some comment

Reply:
We don't relative path anymore. [See](https://github.com/peak/s5cmd/pull/730#issuecomment-2244611229)